.github/workflows/test.yml

```diff
@@ -144,14 +145,54 @@ jobs:
           yarn tailwind-preset build
           yarn components build
       - name: Test Components
-        run: yarn components test-ct --shard ${{ matrix.shard }}
-      - name: Upload Results
-        uses: actions/upload-artifact@v4
+        run: yarn components test-ct --shard=${{ matrix.shardIndex }}/${{ matrix.shardTotal }} --reporter=blob
+      - name: Upload blob report artifacts
         if: failure()
+        uses: actions/upload-artifact@v4
+        with:
+          name: blob-report-${{ matrix.shardIndex }}
+          path: packages/orbit-components/blob-report
+          retention-days: 1
+
+  test-ct-report:
+    name: Visual Regression Tests Report
+    runs-on: ubuntu-latest
+    if: always()
```

Comment:
I wonder if there's a _cleaner_ why to run this job **only** when `test-ct` fails instead of always and do the checks inside.

Have you considered something like this? (untested)

```yml
if: ${{ needs.test-ct.result == 'failure' }}
```

Disregard this comment if you find it so cumbersome, we can tackle this in future iterations.

Reply:
On the official docs, they recommend using `always()`. I have tried several other approaches and they all failed in some case. I did not try that one. I can give it a quick try

Reply:
So it works for when all tests pass. Will now test with failing tests

Reply:
This is cool, let's see 🤞 

![Uploading image.png…]()

Reply:
As discussed IRL, `if: ${{ always() && needs.test-ct.result == 'failure' }}` does the trick. So we will go with that and the simplification of the `run` part, removing the verification for the artifacts existance